src/Adapter/Store/QueryHandler/GetStoreHandler.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Store\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\Store\Exception\StoreException;
+use PrestaShop\PrestaShop\Core\Domain\Store\Exception\StoreNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Store\Query\GetStore;
+use PrestaShop\PrestaShop\Core\Domain\Store\QueryHandler\GetStoreHandlerInterface;
+use PrestaShopException;
+use Store;
+
+class GetStoreHandler implements GetStoreHandlerInterface
```

Comment:
We don't create CQRS queries that return the object directly The purpose of the CQRS architecture is to create a middleware between our domain logic (represented by Queries and Command) which are describing some real actions that are performed in the BO

These handlers return some DTO objects (or QueryResult) that are container of the data, but it doesn't return the legacy object itself. Because the very purpose of the CQRS is to have an abstraction representing the domain logic without any reference to the persistence layer behind it (in this case the ObjectModel `Store`).

Reply:
This handler should return a QueryResult object that contains the store data, and its name should also reflect to the purpose of why you are requesting this data (which maybe you don't have right now since it's a pure CQRS PR without context).

For example, if you are aiming to edit the Store in a form we would have a `GetStoreForEditing` query (with the associated handler) that will return a `StoreForEditing` DTO object containing all the data that you need to prefill your form

Reply:
You can look at the `GetProductForEditing` query to get an idea, although this one is very complex because the product form is huge

Reply:
TLDR; this query should reflect more its purpose, and you need to use DTO as a returned object not the legacy object model itself